### PR TITLE
Add cap_drop to docker_container module

### LINKED ISSUE #29578: docker_container - missing cap_drop
From @gtirloni on 2016-10-17T15:15:06Z
##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

docker_container module
##### ANSIBLE VERSION

ansible 2.1.2.0
##### SUMMARY

docker_container module does not support removing capabilities (i.e. docker --cap-drop option). 

cap_drop and cap_add were added to the docker module in ansible 2.0 but that module is marked as deprecated now.

Copied from original issue: ansible/ansible-modules-core#5286

### PR DESCRIPTION
**Pull request by DavidWittman**
Closes #29578

##### SUMMARY

Adds the option to drop capabilities as well as add them to docker containers.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

docker_container

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.6.0 (devel 0694aca2a0) last updated 2018/02/28 22:17:49 (GMT -500)
  config file = None
  configured module search path = [u'/Users/dwittman/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dwittman/Projects/ansible/lib/ansible
  executable location = /Users/dwittman/Projects/ansible/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/docker/docker_container.py [hunk 1]
```diff
@@ -40,6 +40,11 @@
       - List of capabilities to add to the container.
     default: null
     required: false
+  cap_drop:
+    description:
+      - List of capabilities to drop from the container.
+    default: null
+    required: false
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/docker/docker_container.py]: Add 

```
version_added: 2.6
```

Comment (DavidWittman) [reply lib/ansible/modules/cloud/docker/docker_container.py]: Done, thank you.

**Review by Akasurde — APPROVED** [review]

**Comment by Akasurde** [comment]
@DavidWittman Can you please rebase this branch ?

**Comment by DavidWittman** [comment]
@Akasurde Done.

**Comment by Akasurde** [comment]
Can anyone else take a look into this PR so that we can make this PR rolling?

**Comment by DavidWittman** [comment]
@chouseknecht sorry for the direct notification, but can I bribe you to review this PR?

**Comment by DavidWittman** [comment]
@kassiansun Can you help me find someone to review this PR? Thank you.

**Review by kassiansun — APPROVED** [review]
LGTM

**Comment by DavidWittman** [comment]
@rcarrillocruz @kassiansun Politely requesting a merge or update here. Thank you!

**Comment by kassiansun** [comment]
shipit

**Comment by archmangler** [comment]
Is there any chance of this getting merged in the *near* future? It's really needed to use cap_drop for use-cases like implementing the Docker CIS Security benchmark recommendations?

**Comment by kassiansun** [comment]
We need another `shipit` from module maintainers. The previous active @chouseknecht seems to be a manager now, and the others, I've not seen them for a long time...

**Comment by Akasurde** [comment]
Closing and re-opening for CI trigger.

**Comment by Akasurde** [comment]
@DavidWittman Can you fix the above mention error and let me know ? Thanks.

**Comment by DavidWittman** [comment]
@Akasurde I bumped `version_added` again and squashed some commits. Thank you!

**Comment by Akasurde** [comment]
@DavidWittman Thanks for the contribution. @kassiansun Thanks for the review.

**Pull request merged**